lib/src/data/repositories/database_repository.dart

```diff
@@ -0,0 +1,74 @@
+import '../../core/utils/enums/endpoint_enum.dart';
+import '../../core/utils/resources/data_state.dart';
+import '../../domain/entities/genre_entity.dart';
+import '../../domain/entities/movie_entity.dart';
+import '../../domain/repositories/i_database_repository.dart';
+import '../datasources/local/database.dart';
+
+class DatabaseRepository implements IDatabaseRepository {
+  final AppDatabase database;
+
+  DatabaseRepository(
+    this.database,
+  );
+
+  @override
+  Future<void> saveMovie(
+    MovieEntity movie,
+    EndpointEnum endpoint,
+  ) async {
+    MovieEntity? existingMovie = await getMovieById(movie);
+    if (existingMovie != null) {
+      if (!existingMovie.category.contains(endpoint.name)) {
```

Comment:
You can wrap this condition in one line 
`movie != null && !movie?.category.contains(endpoint.name)`